src/comparator/service_comparator.py

```diff
@@ -119,6 +119,46 @@ def _compareRpcMethods(
                 FindingContainer.addFinding(
                     FindingCategory.METHOD_PAGINATED_RESPONSE_CHANGE, "", msg, True
                 )
+            # 6.8 The method_signature annotation is changed.
+            signatures_original = self._get_signatures(method_original)
+            signatures_update = self._get_signatures(method_update)
+            self._compare_method_signatures(signatures_original, signatures_update)
+
+    def _get_signatures(self, method: MethodDescriptorProto):
+        """Return the signature defined for this method."""
+        return method.options.Extensions[client_pb2.method_signature]
+
+    def _compare_method_signatures(self, signatures_original, signatures_update):
+        def _filter_fields(signatures: [str]) -> [str]:
+            fields = []
+            for sig in signatures:
+                for f in sig.split(','):
+                    if not f:
+                        # Special case for an empty signature
+                        continue
+                    name = f.strip()
+                    fields.append(name)
+            return fields
```

Comment:
Personal opinion nit: this would be more clear as a comprehension.
```python
fields = [field.strip()
          for sig in signatures
          for field in sig.split(",") if field]
```

Reply:
Nice! thanks for the suggestion!